core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.ts

```diff
@@ -410,6 +459,36 @@ export class BaseUser {
   getCurrentUrlWithoutParameters(): string {
     return this.page.url().split('?')[0];
   }
+
+  /**
+   * This function takes a screenshot of the page
+   * If there's no parameter for newPage, it check this.page instead of newPage.
+   * If there's no image named as the given string, it stores the screenshot with the given string in the file __image_snapshots__
+   * Otherwise, it compares the screenshot with the image named as the given string to check if they match.
+   * If they don't match, it generates an image in the file __diff_output__ to show the difference.
```

Comment:
Hm. I can't investigate now, but please file a bug to track if future follow-up work is needed here -- thanks!

Reply:
Without the storeReceivedOnFailure, jest-image-snapshot will only store png files that show both screenshots and the difference in between like this:
![navigateToAboutFoundationPageViaFooter-diff](https://github.com/oppia/oppia/assets/122439444/09fa4b90-c50d-4b59-b1fe-45f92d3c62ee)
With the storeReceivedOnFailure, jest-image-snapshot will also store the failed screenshots like this:
![navigateToAboutFoundationPageViaFooter-received](https://github.com/oppia/oppia/assets/122439444/9761f7bc-d525-450f-b24d-9aa176407dbb)

This might be useful when debugging so I will keep it but I think it doesn't help when the test fails in CI. I will gitignore those folders as well. 
When any screenshot tests fail in CI, it will stop running the rest and we can't see the files. So, I'm thinking if we should disable the screenshot testing part during CI. Another reason for that is the difference between dev mode and prod mode, as you can see in the png files above, I was running and storing all the png files in dev mode and they were failing in prod mode because of the slight difference between dev mode and prod mode. (the word "dev mode" in the bottom left and the absence of the user logo in the top right corner). 

Possible solutions:
1. Store screenshots in both dev mode and prod mode: that will be double the amount of image files and it might not be ideal to push to GitHub, given that we will also store screenshots in mobile viewport
2. Increase the percentage of the threshold that would trigger a test failure: The difference between screenshots in dev mode and in prod mode is around 4%, which are the "dev mode" and the use logo. So adding 4% to the original percentage should allow the tests to pass but it would also accidentally allow other difference as long as the difference are within 4%
3. Storing the screenshots in prod mode instead of dev mode, and only apply the 4% during dev mode: same possible issue as I describe above but at least the tests will pass in CI
4. Disable the screenshot testing in prod mode and update the screenshots regularly. 

@agallop @Akhilesh-max What do yall think? Personally, I would prefer the solution no.3, and then no.4.

Reply:
Can we define different target folders or file names based on whether it's dev or prod mode, and choose the relevant folder or file name based on the mode? I think if we are judicious with not taking too many screenshots, that might be OK.

I think (2) and (3) are a bit handwavy and could cause some errors to be missed, but I'm not too worried as long as reviewers can easily see the diff. So (3) is probably OK too. We could also at some point try to make the "dev mode" less prominent and use a standard user logo for tests, to reduce the skew.

So I think my preference is (1), followed by (3).

Reply:
@agallop @Akhilesh-max WDYT?

Reply:
We discussed the possibility of adding dev move to CI (in addition to prod mode) in the last meeting. Whether or not we are okay with that changes my preferences.

With dev mode in CI.
(1) or (4) are okay with me, with a leaning towards 4 as it seems more maintainable.

Without dev mode in CI.
(3) is my preference. 
For (1) and (4), I am not in favor of pushing screenshots that are not being used by CI. They'll just get forgotten about.